jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -595,7 +595,7 @@ protected boolean passConditionalHeaders(HttpServletRequest request, HttpServlet
                     return false;
                 }
 
-                long ifmsl = request.getDateHeader(HttpHeader.IF_MODIFIED_SINCE.asString());
+                long ifmsl = DateParser.parseDate(ifms);
```

Comment:
Doesn't `request.getDateHeader()` return a long already?  Why return/keep the String then convert it later?

Reply:
Calling `getDateHeader` does another iteration over the fields, which we have already done above.  We don't get the long value above because we don't want to parse it unless we know there is no If-None-Match header, but we won't know that until after we finished the iteration.

So we just remember the String and only convert if we need to.  Thus only a single iteration and only a conversion if we need it.